### PR TITLE
Loadable Apps bring up for RTK8730e

### PR DESCRIPTION
**Pull request by kishore-sn**
Bring up code for loadable apps on RTK8730e chipset. This commit does not include memory protection for loadable apps. Since Armv7a uses MMU instead of MPU, we will make new implementation for memory protection using MMU.

### CODE REVIEW

File: lib/libc/machine/arm/armv7-a/arch_elf.c [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+
+  return OK;
+}
+
+int up_relocateadd(const Elf32_Rela *rel, const Elf32_Sym *sym,
+                   uintptr_t addr)
+{
+  berr("ERROR: RELA relocation not supported\n");
+  return -ENOSYS;
+}
```
Comment (sunghan-chang) [inline lib/libc/machine/arm/armv7-a/arch_elf.c:499]: Let's follow our coding rules.

File: lib/libc/machine/arm/armv7-a/arch_elf.c [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+      return -EINVAL;
+    }
+
+  return OK;
+}
+
+int up_relocateadd(const Elf32_Rela *rel, const Elf32_Sym *sym,
+                   uintptr_t addr)
+{
+  berr("ERROR: RELA relocation not supported\n");
```
Comment (sunghan-chang) [inline lib/libc/machine/arm/armv7-a/arch_elf.c:497]: Do we have `berr`, `binfo` and so on?

File: os/include/sys/syscall_lookup.h [hunk 3]
```diff
@@ -0,0 +1,406 @@
+/****************************************************************************
```
Comment (abhishek-samsung) [inline os/include/sys/syscall_lookup.h:1]: Is this file needed? We have syscall.csv which is used to generate similar mappings.

Comment (abhishek-samsung) [reply os/include/sys/syscall_lookup.h:1]: Looks like we already have the file,
https://github.com/Samsung/TizenRT/blob/master/os/syscall/syscall_lookup.h

File: lib/libc/machine/arm/armv7-a/arch_elf.c (hunk 1 above)
Comment (kishore-sn) [reply lib/libc/machine/arm/armv7-a/arch_elf.c:499]: Done.

File: os/syscall/syscall_names.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/* System call name tables.  This table is indexed by the system call numbers
+ * defined above.  Given the system call number, this table provides the
+ * name of the corresponding system function.
+ */
+
+const char *g_funcnames[SYS_nsyscalls] =
```
Comment (abhishek-samsung) [inline os/syscall/syscall_names.c:97]: I think this file is also present under a different name,

https://github.com/Samsung/TizenRT/blob/ebdd02ca6b5b34e535d696492a0f5ef6732c9c4c/os/syscall/syscall_funclookup.c#L115-L122

could you please check if the new file is necessary?

Comment (kishore-sn) [reply os/syscall/syscall_names.c:97]: `g_funclookup` returns the function address and `g_funcnames` returns the function name as a string. So, both are required. However, there was a mistake in the macro expansion to return the function name. I have fixed it. Thanks.

**Review by abhishek-samsung — APPROVED** [review]

File: build/configs/rtl8730e/Make.defs [hunk 5]
```diff
@@ -121,6 +121,9 @@ ifeq ($(CONFIG_FRAME_POINTER),y)
 endif
 
 ARCHCFLAGS += -fno-builtin 
+ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
```
Comment (Tirtha-Samsung) [inline build/configs/rtl8730e/Make.defs:124]: The PR, Loadable Apps bring up for RTK8730e, looks good to me.

**Review by sunghan-chang — APPROVED** [review]
